src/message-stream.ts

```diff
@@ -301,8 +298,14 @@ export class MessageStream extends PassThrough {
    */
   private _onEnd(stream: PullStream, status: StatusObject): void {
     this._removeStream(stream);
-
-    if (RETRY_CODES.includes(status.code)) {
+    if (Date.now() - this._timeSinceFirstError > this._options.timeout!) {
```

Comment:
So to clarify, this will wait to see if the stream was open for more than 30 seconds (or whatever it was set to) and if so we consider it a success?

Reply:
Its actually checks if there are any errors in stream try creating new stream for 30 sec or max retries and fail afterwards.